### PR TITLE
Accounting localisations for Ghana and Nigeria by erpSOFTapp

### PR DESCRIPTION
**Pull request by erpsoftapp**
Description of the issue/feature this PR addresses: Accounting localisations for Ghana and Nigeria

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_gh/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+===========================================================================
+
+* Chart of Accounts.
+
+    """,
+    'author': 'erpSOFTapp',
+    'license': 'AGPL-3',
+    'depends': [
+        'account',
+        'account_accountant',
```
Comment (jco-odoo) [inline addons/l10n_gh/__manifest__.py:18]: just account will do

File: addons/l10n_gh/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                'base'
+
+    ],
+    'data': [
+        'data/l10n_gh_chart_data.xml',
+        'data/account.account.template.csv',
+        'data/account.chart.template.csv',
+        'data/account.tax.group.csv',
+        'data/account_tax_data.xml',
+        'data/res.country.state.csv',
```
Comment (jco-odoo) [inline addons/l10n_gh/__manifest__.py:28]: better in base and only if it is really needed in the postal addresses

File: addons/l10n_gh/data/account_tax_data.xml [hunk 3]
```diff
@@ -0,0 +1,465 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <!--VATs-->
+
+
+    <record id="vat125s" model="account.tax.template">
+        <field name="chart_template_id" ref="l10n_gh"/>
+        <field name="name">VAT @12.5%</field>
```
Comment (jco-odoo) [inline addons/l10n_gh/data/account_tax_data.xml:9]: What is with the @

File: addons/l10n_gh/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    'category': 'Localization/Account Charts',
+    'description': """
+This is the base module to manage the accounting chart for Ghana.
+===========================================================================
+
+* Chart of Accounts.
+
+    """,
+    'author': 'erpSOFTapp',
+    'license': 'AGPL-3',
```
Comment (jco-odoo) [inline addons/l10n_gh/__manifest__.py:15]: Just leave it out, See license file...

File: addons/l10n_gh/data/account.tax.group.csv [hunk 5]
```diff
@@ -0,0 +1,4 @@
+id,name
+tax_group_vat_gh, Value Added Tax
```
Comment (jco-odoo) [inline addons/l10n_gh/data/account.tax.group.csv:2]: Only one VAT rate of 12.5% ?

File: addons/l10n_gh/data/account.tax.group.csv [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+tax_group_wht_gh, Withholding Tax
```
Comment (jco-odoo) [inline addons/l10n_gh/data/account.tax.group.csv:3]: No need to group by different rates of withholding?

**Review by jco-odoo — CHANGES_REQUESTED** [review]
For the withholding, is it all right that it is withdrawn from the total amount of the invoice then?  
Ideally, we should have this together with tax report (VAT return) and P&L/Balance sheet so it is fully usable.

File: addons/l10n_gh/__manifest__.py (hunk 1 above)
Comment (erpsoftapp) [reply addons/l10n_gh/__manifest__.py:18]: Yes WHT is withdrawn from the total amount on the invoice for these 2 countries.
When you say have it with tax report and P&L balance sheet are you referring to a separate xx_report module to incorporate these..can you clarify 

We will remove account_accountant

File: addons/l10n_gh/__manifest__.py (hunk 2 above)
Comment (erpsoftapp) [reply addons/l10n_gh/__manifest__.py:28]: Yes for Ghana the regions in the states CSV is needed in the postal addresses
When you say better in base can you clarify that aspect?

File: addons/l10n_gh/data/account_tax_data.xml (hunk 3 above)
Comment (erpsoftapp) [reply addons/l10n_gh/data/account_tax_data.xml:9]: Will remove the @ from the names

File: addons/l10n_gh/__manifest__.py (hunk 4 above)
Comment (erpsoftapp) [reply addons/l10n_gh/__manifest__.py:15]: will remove license section

File: addons/l10n_gh/data/account.tax.group.csv (hunk 5 above)
Comment (erpsoftapp) [reply addons/l10n_gh/data/account.tax.group.csv:2]: Just another VAT rate to add will do so

File: addons/l10n_gh/__manifest__.py (hunk 2 above)
Comment (jco-odoo) [reply addons/l10n_gh/__manifest__.py:28]: There is a res.country.state.csv file in the base module (odoo/odoo) already.  It is better that everyone has them then.

File: addons/l10n_gh/data/account.tax.group.csv (hunk 6 above)
Comment (erpsoftapp) [reply addons/l10n_gh/data/account.tax.group.csv:3]: Will confirm and see if grouping can be used

File: addons/l10n_gh/__manifest__.py (hunk 2 above)
Comment (erpsoftapp) [reply addons/l10n_gh/__manifest__.py:28]: Ok will you take it out of this module and move it to base then?

File: addons/l10n_gh/__manifest__.py (hunk 1 above)
Comment (jco-odoo) [reply addons/l10n_gh/__manifest__.py:18]: No need to specify base below as well.  (as account depends on base)  

The tax report can already be defined in this module.  In the repartition_line_ids, you can add the plus/minus_tag_ids to refer to the tax report line that needs that base/tax amount added/subtracted.  At least for this VAT return the advantage is that we check if the necessary taxes are defined, ...  I see e.g. no fiscal positions and taxes for export. 

For the P&L and balance sheet, that would be through a l10n_gh_reports in enterprise.

**Review by jco-odoo — CHANGES_REQUESTED** [review]
Hi @erpsoftapp ,  I am a bit surprised myself we did not react on this earlier.  Do you have some possibility to finish this together?  A tax report would be nice, but we could merge like this with the minimum changes as well.  Most comments for Ghana, apply for Nigeria as well.

**Comment by qdp-odoo** [comment]
@Laura-labe look what I found! There's some dust here and there but it looks like it can be a good starting point :)

**Comment by C3POdoo** [comment]
Dear @erpsoftapp,

Thank you for your contribution but the version 14.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**